Report Module - Grouping Opportunities by Assigned Users displays user ID instead of Name

<!--- Provide a general summary of the issue in the **Title** above -->
<!--- Before you open an issue, please check if a similar issue already exists or has been closed before. --->
<!--- If you have discovered a security risk please report it by emailing security@suitecrm.com. This will be delivered to the product team who handle security issues. Please don't disclose security bugs publicly until they have been handled by the security team. --->

#### Issue
<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->
THIS IS AFTER AN UPGRADE FROM 7.8.11 TO 7.10.1
When creating a Report of Opportunities, and setting the 'Main Group' to 'Assigned to', the group header displays the users id instead of their name. It also breaks PDF export, in that the PDF only displays the user ID headers and no content under them.

#### Expected Behavior
<!--- Tell us what should happen -->
The group header should show the User's Name.

#### Actual Behavior
<!--- Tell us what happens instead -->
<!--- Also please check relevant logs (suitecrm.log, php error.log etc.) -->
The suitecrm.log file says:
[ERROR] SugarBean.load_relationships, Null relationship name passed.
This error is repeated for each field included.

#### Possible Fix
<!--- Not obligatory, but suggest a fix or reason for the bug -->

#### Steps to Reproduce
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug include code to reproduce, if relevant -->
1. Create Report for the Opportunities Module
2. Select "Opportunity Name" and "Assigned to" fields
3. Set the "Main Group" to "Assigned to"
4.

#### Context
<!--- How has this bug affected you? What were you trying to accomplish? -->
<!--- If you feel this should be a low/medium/high priority then please state so -->

#### Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* SuiteCRM Version used: 7.10.1
* Browser name and version: Chrome 64.0.3282.186 (Official Build) (64-bit)
* Environment name and version (e.g. MySQL, PHP 7): 10.0.33-MariaDB, PHP 7.0.25-0ubuntu0.16.04.1
* Operating System and version: Ubuntu 16.04

--- a/modules/AOR_Reports/AOR_Report.php
+++ b/modules/AOR_Reports/AOR_Report.php
@@ -553,19 +553,33 @@ class AOR_Report extends Basic
                     $groupDisplay = $app_strings['LBL_NONE'];
                 }
 
+                 // Fix #5427 If download pdf then not use tab-content and add css inline to work with mpdf
+                $pdf_style = "";
+                $action = $_REQUEST['action'];
+                if ($action == 'DownloadPDF') {
+                    $pdf_style = "background: #333 !important; color: #fff !important; margin-bottom: 0px;";
+                }
+
                 $html .= '<div class="panel panel-default">
-                            <div class="panel-heading ">
+                            <div class="panel-heading" style="' . $pdf_style . '">
                                 <a class="" role="button" data-toggle="collapse" href="#detailpanel_report_group_' . $groupValue . '" aria-expanded="false">
                                     <div class="col-xs-10 col-sm-11 col-md-11">
                                         ' . $groupDisplay . '
                                     </div>
                                 </a>
-                            </div>
-                            <div class="panel-body panel-collapse collapse in" id="detailpanel_report_group_' . $groupValue . '">
+                            </div>';
+                if ($action != 'DownloadPDF') {
+                    $html .= '<div class="panel-body panel-collapse collapse in" id="detailpanel_report_group_' . $groupValue . '">
                                 <div class="tab-content">';
+                }
+                else {
+                    $html .= '</div>';
+                }
+
 
                 $html .= $this->build_report_html($offset, $links, $groupValue, create_guid(), $extra);
-                $html .= '</div></div></div>';
+                $html .= ($action == 'downloadPDF') ? '' : '</div></div></div>';
+                // End
 
             }
         }
@@ -614,7 +628,16 @@ class AOR_Report extends Basic
             }
         }
 
-        $html='<div class="list-view-rounded-corners">';
+          // Fix #5427
+        $report_style = '';
+        $thead_style = '';
+        if ($_REQUEST['action'] == 'DownloadPDF') {
+            $report_style = 'margin-top: 0px;';
+            $thead_style = 'background: #919798; color: #fff';
+        }
+        $html = '<div class="list-view-rounded-corners" style="' . $report_style . '">';
+        //End
+        
         $html.='<table id="report_table_'.$tableIdentifier.$group_value.'" cellpadding="0" cellspacing="0" width="100%" border="0" class="list view table-responsive aor_reports">';
 
         $sql = "SELECT id FROM aor_fields WHERE aor_report_id = '" . $this->id . "' AND deleted = 0 ORDER BY field_order ASC";
@@ -659,7 +682,9 @@ class AOR_Report extends Basic
 
 
             if ($fields[$label]['display']) {
-                $html .= "<th scope='col'>";
+                 // Fix #5427
+                $html .= "<th scope='col' style='{$thead_style}'>";
+                // End
                 $html .= "<div>";
                 $html .= $field->label;
                 $html .= "</div></th>";
@@ -835,8 +860,18 @@ class AOR_Report extends Basic
             $field->retrieve($row['id']);
 
             if ($field->field_function != 'COUNT' || $field->format != '') {
-                $moduleFieldByGroupValues[] = $group_value;
+                // Fix grouping on assignment displays ID and not name #5427
+                $report_bean = BeanFactory::getBean($this->report_module);
+                $field_def = $report_bean->field_defs[$field->field];
+                if ($field_def['type'] == 'relate' && isset($field_def['id_name'])) {
+                    $related_bean = BeanFactory::getBean($field_def['module']);
+                    $related_bean->retrieve($group_value);
+                    $moduleFieldByGroupValues[] = ($related_bean instanceof Person) ? $related_bean->full_name : $related_bean->name;
+                } else {
+                    $moduleFieldByGroupValues[] = $group_value;
+                }
                 continue;
+                // End
             }
 
             $path = unserialize(base64_decode($field->module_path));

